modules/custom/az_finder/src/Form/AZFinderSettingsForm.php

```diff
@@ -0,0 +1,412 @@
+      'overrides',
+      'select_view_display_container',
+      'select_view_display',
+    ]);
+    [$view_id, $display_id] = explode(':', $selected_view_display);
+
+    // Prepare the configuration key.
+    $config_name = "az_finder.tid_widget.$view_id.$display_id";
+
+    // Initialize or load existing configuration.
+    $config = $this->configFactory->getEditable($config_name);
+
+    // Save the configuration.
+    $config->save();
+
+    // Optionally, provide feedback or perform additional actions.
+    $this->messenger()->addMessage($this->t('Override created for @view_display.', ['@view_display' => $selected_view_display]));
+  }
+
+  /**
+   * Ajax callback for the override button.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   *
+   * @return array
+   *   The updated overrides container.
+   */
+  public function ajaxAddOverride(array &$form, FormStateInterface $form_state): array {
+    // Get the selected option from the form state.
+    $selected_option = $form_state->getValue([
+      'az_finder_tid_widget',
+      'overrides',
+      'select_view_display_container',
+      'select_view_display',
+    ]);
+
+    // Split the selected option into view_id and display_id.
+    [$view_id, $display_id] = explode(':', $selected_option);
+    $override = [
+      'view_id' => $view_id,
+      'display_id' => $display_id,
+      'origin' => 'session',
+    ];
+
+    // Ensure the overrides array is present in the form state.
+    $overrides = $form_state->getValue(['az_finder_tid_widget', 'overrides']) ?? [];
+    // Update the overrides with the new override.
+    $overrides["$view_id:$display_id"] = $override;
+    $form_state->setValue(['az_finder_tid_widget', 'overrides'], $overrides);
+    // Add the override section to the form.
+    $this->addOverrideSection($form, $form_state, $override);
+    // Set the rebuild flag to ensure the form is rebuilt.
+    $form_state->setRebuild(TRUE);
+
+    // Return the updated overrides container.
+    return $form['az_finder_tid_widget']['overrides'];
+  }
+
+  /**
+   * Add an override section to the form.
+   *
+   * @param array $form
+   *   The form array.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state object.
+   * @param array $override
+   *   The override data.
+   *
+   * @return ?array
+   *   Return the override section or null.
+   */
+  public function addOverrideSection(array &$form, FormStateInterface $form_state, array $override): ?array {
+    $key = "{$override['view_id']}:{$override['display_id']}";
+    $view_id = $override['view_id'];
+    $display_id = $override['display_id'];
+    if ($key !== ':' && !isset($form['az_finder_tid_widget']['overrides'][$key])) {
+      $form['az_finder_tid_widget']['configure_overrides'] = [
+        '#type' => 'item',
+        '#title' => $this->t('Configure Applied Overrides'),
```

Comment:
This is inside a loop, see comment on line 174 of modules/custom/az_finder/src/Form/AZFinderSettingsForm.php